packages/terra-menu/src/_MenuContent.jsx

```diff
@@ -143,12 +162,34 @@ class MenuContent extends React.Component {
   }
 
   render() {
+    let index = -1;
     const items = React.Children.map(this.props.children, (item) => {
+      let onClick = item.props.onClick;
+      let newItem = item;
+
       if (item.props.subMenuItems && item.props.subMenuItems.length > 0) {
-        return React.cloneElement(item, { onClick: this.wrapOnClick(item), onKeyDown: this.wrapOnKeyDown(item) });
+        onClick = this.wrapOnClick(item);
+      }
+
+      if (item.props.text) {
+        index += 1;
+        const onKeyDown = this.wrapOnKeyDown(item, index);
+        const isActive = this.state.focusIndex === index;
+
+        newItem = React.cloneElement(item, {
+          onClick,
+          onKeyDown,
+          isActive,
+        });
+      } else if (item.props.children) {
+        const children = React.Children.map(item.props.children, (child) => {
+          index += 1;
```

Comment:
Could we move index+=1 and isActive outside of if logic?

Reply:
No, that logic shouldn't be applied to Divider elements, and in the case of ItemGroups the isActive should be applied to it's children, not the ItemGroup.

Reply:
oh right. At first glance I didn't realize that. Maybe we should add a comment?